### PR TITLE
4. Adding Support For VPA and Auto Mode in CreateExperiment [KRUIZE-VPA Integration]

### PR DESCRIPTION
**Pull request by shekhar316**
## Description

This PR adds `auto` and `recreate` modes required for VPA in createExperiment for Local Monitoring. 

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?
Tested on both the ResourceHub cluster.

Test Configuration

Kubernetes clusters tested on: ResourceHub (OpenShift)

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

### CODE REVIEW

**Review by bharathappali — APPROVED** [review]
LGTM

File: src/main/java/com/autotune/analyzer/kruizeObject/ExperimentUseCaseType.java [hunk 1]
```diff
@@ -43,6 +43,10 @@ public ExperimentUseCaseType(KruizeObject kruizeObject) throws Exception {
                 setLocal_monitoring(true);
             } else if (kruizeObject.getMode().equalsIgnoreCase(AnalyzerConstants.EXPERIMENT)) {
                 setLocal_experiment(true);
+            } else if (kruizeObject.getMode().equalsIgnoreCase(AnalyzerConstants.RECREATE)) {
+                setLocal_monitoring(true);
+            } else if (kruizeObject.getMode().equalsIgnoreCase(AnalyzerConstants.AUTO)) {
+                setLocal_monitoring(true);
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/kruizeObject/ExperimentUseCaseType.java:49]: Maybe a switch case would be more appropriate here

File: src/main/java/com/autotune/analyzer/kruizeObject/ExperimentUseCaseType.java [hunk 2]
```diff
@@ -39,12 +39,22 @@ public ExperimentUseCaseType(KruizeObject kruizeObject) throws Exception {
...
+                case AnalyzerConstants.MONITOR:
+                    setLocal_monitoring(true);
+                    break;
+
+                case AnalyzerConstants.EXPERIMENT:
+                    setLocal_experiment(true);
+                    break;
+
+                case AnalyzerConstants.RECREATE:
+                case AnalyzerConstants.AUTO:
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/kruizeObject/ExperimentUseCaseType.java:52]: Please combine this with MONITOR as currently they are all the same

Comment (shekhar316) [reply src/main/java/com/autotune/analyzer/kruizeObject/ExperimentUseCaseType.java]: Sure dino! I have updated the code to use `switch` `case` block and combined the `RECREATE` and `AUTO` cases with the `MONITOR` case.

**Comment by dinogun** [comment]
`auto` and `recreate` are not `monitor` modes technically so we need to change this but that requires a design discussion. We can do this as a short term solution for testing VPA. Longer term these two modes have to have their own separate category.

**Review by dinogun — APPROVED** [review]
LGTM for VPA alpha support

**Pull request merged**